dashboard/app/controllers/lessons_controller.rb

```diff
@@ -43,13 +43,17 @@ def edit
 
   # PATCH/PUT /lessons/1
   def update
-    resources = (lesson_params['resources'] || []).map {|key| Resource.find_by_key(key)}
-    @lesson.resources = resources.compact
-    @lesson.update!(lesson_params.except(:resources, :objectives))
-    @lesson.update_activities(JSON.parse(params[:activities])) if params[:activities]
-    @lesson.update_objectives(JSON.parse(params[:objectives])) if params[:objectives]
+    ActiveRecord::Base.transaction do
+      resources = (lesson_params['resources'] || []).map {|key| Resource.find_by_key(key)}
```

Comment:
Should these non-update statements be outside of the transaction or does it not matter?

Reply:
yes, probably best to do as little as possible inside the transaction!